src/snowflake/snowpark/functions.py

```diff
@@ -6581,6 +6611,35 @@ def check_xml(col: ColumnOrName, _emit_ast: bool = True) -> Column:
     return builtin("check_xml", _emit_ast=_emit_ast)(c)
 
 
+@publicapi
+def json_tuple(col: ColumnOrName, *fields: str, _emit_ast: bool = True) -> List[Column]:
+    """Create new rows for a json column according to given json field.
+
+    Example::
+
+        >>> from snowflake.snowpark.functions import json_tuple
+        >>> data = [("1", '''{"key1": "value1", "key2": "value2"}'''), ("2", '''{"key1": "value2"}''')]
+        >>> df = session.createDataFrame(data, ("id", "jstring"))
+        >>> df.select(df.id, json_tuple(df.jstring, 'key1', 'key2')).show()
+        --------------------------
+        |"ID"  |"C0"    |"C1"    |
+        --------------------------
+        |1     |value1  |value2  |
+        |2     |value2  |NULL    |
+        --------------------------
+        <BLANKLINE>
+    """
+    c = _to_col_if_str(col, "json_tuple")
+    return [
+        json_extract_path_text(
+            parse_json(c, _emit_ast=False),
+            lit(field, _emit_ast=False),
+            _emit_ast=False,
+        ).as_(f"c{i}", _emit_ast=False)
+        for i, field in enumerate(fields)
+    ]
```

Comment:
I don't think this API would emit any AST currently. 

Modeling this API with the AST would still be more involved though, as this is the first Snowpark function that returns a `List[Column]` AFAIK. Usually we rely on the returned `Column` object storing its own AST in the `._ast` field.

Is there any way the return type here can be a `Column` object? This would allow the AST to be constructed via `build_builtin_fn_apply` and assigned to the `Column` instance to be returned.